h2o-core/src/main/java/water/server/ServletService.java

```diff
@@ -0,0 +1,35 @@
+package water.server;
+
+import javax.servlet.http.HttpServlet;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public final class ServletService {
+
+  public static final ServletService INSTANCE = new ServletService();
+
+  private final ServiceLoader<ServletProvider> _loader;
+
+  private ServletService() {
+    _loader = ServiceLoader.load(ServletProvider.class);
+  }
+
+  /**
+   * Locates all available non-core primitives of the Rapid language.
+   * @return list of Rapid primitives
+   */
+  public LinkedHashMap<String, Class<? extends HttpServlet>> getAllServlets() {
+    List<ServletProvider> providers;
+    synchronized (_loader) {
```

Comment:
Why the block is synchronized if `providers` is just a variable with the method.

Reply:
Because iterating over iterator provided by service loader is not thread-safe (and this class is a singleton).